kamon-metrics/src/main/scala/com/ovoenergy/effect/Metrics.scala

```diff
@@ -0,0 +1,34 @@
+package com.ovoenergy.effect
+
+import cats.FlatMap
+import cats.effect.Sync
+import com.ovoenergy.effect.Metrics.Metric
+import kamon.Kamon
+
+import scala.language.higherKinds
+
+/**
+ * A type class representing the ability to push metrics
+ * within some effect type F
+ */
+trait Metrics[F[_]] {
+  def counter(metric: Metric): F[Long => F[Unit]]
+}
+
+object Metrics {
+
+  case class Metric(name: String, tags: Map[String, String])
+
+  def apply[F[_]: Metrics]: Metrics[F] = implicitly
+
+  /**
+   * An instance of metrics for a Sync[F] which wraps the underlying calls
+   */
+  //noinspection ConvertExpressionToSAM
```

Comment:
?

Reply:
otherwise it will complain because we have only one function so we could avoid writing the signature and give just the body.  if people prefer the other way I am happy to change

Reply:
yeah if its possible to write it in the shorter way I'd do it, sometimes though IntelliJ emits the warning for things that can't be converted, though this looks okay (one method, no type params)